components/cdc/src/endpoint.rs

```diff
@@ -515,81 +559,70 @@ impl Initializer {
-                            Ok(res) => res,
-                            Err(e) => {
-                                error!("cdc scan entries failed"; "error" => ?e);
-                                // TODO: record in metrics.
-                                let deregister = Deregister::Downstream {
-                                    region_id,
-                                    downstream_id,
-                                    conn_id,
-                                    err: Some(e),
-                                };
-                                if let Err(e) = sched.schedule(Task::Deregister(deregister)) {
-                                    error!("schedule cdc task failed"; "error" => ?e);
-                                }
-                                return Ok(());
-                            }
-                        };
-                    // If the last element is None, it means scanning is finished.
-                    if let Some(None) = entries.last() {
-                        done = true;
-                    }
-                    debug!("cdc scan entries"; "len" => entries.len());
-                    let scanned = Task::IncrementalScan {
+        let mut resolver = if self.build_resolver {
+            Some(Resolver::new(region_id))
+        } else {
+            None
+        };
+
+        fail_point!("cdc_incremental_scan_start");
+
+        let start = Instant::now_coarse();
+        // Time range: (checkpoint_ts, current]
+        let current = TimeStamp::max();
+        let mut scanner = ScannerBuilder::new(snap, current, false)
+            .range(None, None)
+            .build_delta_scanner(self.checkpoint_ts)
+            .unwrap();
+        let mut done = false;
+        while !done {
+            let entries = match Self::scan_batch(&mut scanner, self.batch_size, resolver.as_mut()) {
+                Ok(res) => res,
+                Err(e) => {
+                    error!("cdc scan entries failed"; "error" => ?e);
+                    // TODO: record in metrics.
+                    let deregister = Deregister::Downstream {
                         region_id,
                         downstream_id,
-                        entries,
+                        conn_id,
+                        err: Some(e),
                     };
-                    if let Err(e) = sched.schedule(scanned) {
-                        error!("schedule task failed"; "error" => ?e);
-                        return Ok(());
+                    if let Err(e) = self.sched.schedule(Task::Deregister(deregister)) {
+                        error!("schedule cdc task failed"; "error" => ?e);
                     }
+                    return;
                 }
+            };
+            // If the last element is None, it means scanning is finished.
+            if let Some(None) = entries.last() {
+                done = true;
+            }
+            debug!("cdc scan entries"; "len" => entries.len());
```

Comment:
Could you please add region_id to the logs in this function?

Reply:
There's some other logs in the function. Could you please also add region_id to them?